core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -61,6 +61,7 @@
     private String queryExecutionPolicy = "all-at-once";
     private Duration queryMaxRunTime = new Duration(100, TimeUnit.DAYS);
     private Duration queryMaxExecutionTime = new Duration(100, TimeUnit.DAYS);
+    private Duration queryMaxPlanningTime = new Duration(10, TimeUnit.MINUTES);
```

Comment:
Is this limit reasonable?

Reply:
How does this differ/relate to the optimizer timeout config? Does `queryMaxPlanningTime` include time from admission until split creation?

10 mins seems like a very generous planning timeout. FWIW, you will hit optimizer timeout after ~3min anyway (default).

Reply:
Is this limit safe to introduce? i.e. can it break some legitimate workloads that work correctly today?

cc @martint @electrum @phd3

Reply:
Maybe we should make it infinite by default?

Reply:
10 minutes is plenty of time. It's extremely unlikely for any well-behaving system to be taking that long to plan.